3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js

```diff
@@ -1,35 +1,11 @@
-/*------------------------------------------------------------------------------
-Full description at: https://github.com/HackYourFuture/Homework/tree/main/3-UsingAPIs/Week1#exercise-4-throw-the-dice-for-a-poker-dice-game
-
-For this exercise you should do the following:
-  - Refactor the `rollDice()` function to throw five dice in one go, by 
-    using `.map()` on the `dice` array to create an array of promises for use 
-    with `Promise.all()`.
-  - A successful (i.e. resolved) throw should output a message similar to: 
-      Resolved! [ 'JACK', 'QUEEN', 'QUEEN', 'NINE', 'JACK' ]
-  - An unsuccessful (i.e. rejected) throw should output a message similar to:
-      Rejected! Die 3 rolled off the table.
-
-The provided rollDie() function logs the value of a die as it rolls, 
-time-stamped with the time of day (with millisecond accuracy) to the console. 
-Once you have successfully completed this exercise you will notice that the 
-intermediate messages are output in bursts of up to five at a time as the dice 
-finish rolling asynchronously.
-
-You may also notice that, in the case of a rejected promise, dice that have not
-yet finished their roll continue to do so. 
-Can you explain why? Please add your answer as a comment to the end of the 
-exercise file.
-------------------------------------------------------------------------------*/
-
-// The line below makes the rollDie() function available to this file.
-// Do not change or remove it.
 const rollDie = require('../../helpers/pokerDiceRoller');
 
 function rollDice() {
-  // TODO Refactor this function
   const dice = [1, 2, 3, 4, 5];
-  return rollDie(1);
+  const proms = dice.map((die) => {
```

Comment:
There is nothing to be gained from abbreviations like this. Express yourselves in the clearest terms possible. Abbreviations are okay if their meaning is commonly understood, e.g. `err` for `error`, `arr` for `array`. `proms` does not fall in that category. If in doubt, don't abbreviate.

```suggestion
  const promises = dice.map((die) => {
```